pgjdbc/src/main/java/org/postgresql/core/CommandCompleteParser.java

```diff
@@ -0,0 +1,107 @@
+/*
+ * Copyright (c) 2018, PostgreSQL Global Development Group
+ * See the LICENSE file in the project root for more information.
+ */
+
+package org.postgresql.core;
+
+import org.postgresql.util.GT;
+import org.postgresql.util.PSQLException;
+import org.postgresql.util.PSQLState;
+
+/**
+ * Parses {@code oid} and {@code rows} from a {@code CommandComplete (B)} message (end of Execute).
+ */
+public final class CommandCompleteParser {
+  private long oid;
+  private long rows;
+
+  public CommandCompleteParser() {
+  }
+
+  public long getOid() {
+    return oid;
+  }
+
+  public long getRows() {
+    return rows;
+  }
+
+  void set(long oid, long rows) {
+    this.oid = oid;
+    this.rows = rows;
+  }
+
+  /**
+   * Parses {@code CommandComplete (B)} message.
+   * Status is in the format of "COMMAND OID ROWS" where both 'OID' and 'ROWS' are optional
+   * and COMMAND can have spaces within it, like CREATE TABLE.
+   *
+   * @param status COMMAND OID ROWS message
+   * @throws PSQLException in case the status cannot be parsed
+   */
+  public void parse(String status) throws PSQLException {
+    // Assumption: command neither starts nor ends with a digit
+    if (!Parser.isDigitAt(status, status.length() - 1)) {
+      set(0, 0);
```

Comment:
Should return here, right?

Reply:
right

Reply:
apparently there isn't a test case for this :)

Reply:
It is not really valid, but something like CREATE 2ABLE would have failed previously.